src/components/StoreModifier.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { ReactReduxContext } from './Context'
+
+export function createStoreModifier(Context = ReactReduxContext) {
+  class StoreModifier extends React.Component {
+    constructor(props, context) {
+      super(props, context)
+
+      this.firstContext = context
+
+      const { store, storeState } = this.context
+      const modifyStore = this.props.modifyStore || this.context.modifyStore
+
+      // Create storeProxy
+      let newRootReducer
+      const replaceReducerSpy = (reducer, ...args) => {
+        newRootReducer = reducer
+        return store.replaceReducer(reducer, ...args)
+      }
+      const storeProxy = { ...store, replaceReducer: replaceReducerSpy }
```

Comment:
Should this also include a `modifyStore` proxy to allow modifier from calling other modifiers?

```js
let storeProxy

const modifyStoreProxy = (modification) => modifyStore(modification, storeProxy, storeState)

storeProxy = { ...store, replaceReducer: replaceReducerSpy: modifyStore: modifyStoreProxy }
```

```js
function dynamicReducers(modification, storeProxy, state) {
  // whatever
}

function addAdminReducer(modification, storeProxy, state) {
  if (state.user.isAdmin) {
    storeProxy.modifyStore({ reducers: [adminReducer] })
  }
}

return (
    <Provider store={store} storeModifier={combineModifiers({ reducers: dynamicReducers, addAdminReducer })}>...</Provider>
);
```

Looking at that above example, I can see why we may not want to use a `combineReducers` like API to solve the namespacing issue.  Not all modifiers will necessarily require a slice of the `modifications` object.

Reply:
Hmm, that's interesting. If we want modifiers to be truly modular and composable I think we'd need something like this. If we do add the modifiers-concept to Redux directly, that function will probably already exist on the store.